pkg/applier/inventory/inventory_converter.go

```diff
@@ -0,0 +1,201 @@
+		}
+		objMetadata := object.ObjMetadata{
+			Name:      strings.TrimSpace(name),
+			Namespace: strings.TrimSpace(namespace),
+			GroupKind: schema.GroupKind{
+				Group: strings.TrimSpace(group),
+				Kind:  strings.TrimSpace(kind),
+			},
+		}
+		klog.V(4).Infof("converting to ObjMetadata: %s", objMetadata)
+		unstructuredInventory.ObjectRefs = append(unstructuredInventory.ObjectRefs, objMetadata)
+	}
+	return unstructuredInventory, nil
+}
+
+// InventoryToUnstructured converts a UnstructuredInventory to a ResourceGroup
+// object, using the current ResourceGroup object as a base.
+func (ic *unstructuredInventoryConverter) InventoryToUnstructured(fromObj *unstructured.Unstructured, toInv *inventory.UnstructuredInventory) (*unstructured.Unstructured, error) {
+	if fromObj == nil {
+		return nil, fmt.Errorf("unstructured object is nil")
+	}
+	if toInv == nil {
+		return nil, fmt.Errorf("UnstructuredInventory object is nil")
+	}
+	objStatusMap := map[object.ObjMetadata]actuation.ObjectStatus{}
+	for _, s := range toInv.GetObjectStatuses() {
+		objStatusMap[inventory.ObjMetadataFromObjectReference(s.ObjectReference)] = s
+	}
+	klog.V(4).Infof("converting UnstructuredInventory to ResourceGroup object")
+	klog.V(4).Infof("creating list of %d resources", len(toInv.GetObjectRefs()))
+	var objs []interface{}
+	for _, objMeta := range toInv.GetObjectRefs() {
+		klog.V(4).Infof("converting to object reference: %s", objMeta)
+		objs = append(objs, map[string]interface{}{
+			"group":     objMeta.GroupKind.Group,
+			"kind":      objMeta.GroupKind.Kind,
+			"namespace": objMeta.Namespace,
+			"name":      objMeta.Name,
+		})
+	}
+	klog.V(4).Infof("Creating list of %d resource statuses", len(toInv.GetObjectRefs()))
+	var objStatus []interface{}
+	for _, objMeta := range toInv.GetObjectRefs() {
+		status, found := objStatusMap[objMeta]
+		if found {
+			klog.V(4).Infof("converting to object status: %s", objMeta)
+			objStatus = append(objStatus, map[string]interface{}{
+				"group":     objMeta.GroupKind.Group,
+				"kind":      objMeta.GroupKind.Kind,
+				"namespace": objMeta.Namespace,
+				"name":      objMeta.Name,
+				"status":    "Unknown",
+				"strategy":  status.Strategy.String(),
+				"actuation": status.Actuation.String(),
+				"reconcile": status.Reconcile.String(),
+			})
+		}
+	}
+
+	toObj := fromObj.DeepCopy()
+	// Set Config Sync specific metadata on the object
+	core.SetLabel(toObj, metadata.ManagedByKey, metadata.ManagedByValue)
+	core.SetLabel(toObj, metadata.SyncNamespaceLabel, ic.syncNamespace)
+	core.SetLabel(toObj, metadata.SyncNameLabel, ic.syncName)
+	core.SetLabel(toObj, metadata.SyncKindLabel, ic.syncKind)
+	core.SetAnnotation(toObj, metadata.ResourceManagementKey, metadata.ResourceManagementEnabled)
+	core.SetAnnotation(toObj, metadata.StatusModeKey, ic.statusMode)
+
+	if len(objs) == 0 { // Unset resource fields
+		klog.V(4).Infoln("clearing inventory resources")
+		unstructured.RemoveNestedField(toObj.UnstructuredContent(),
```

Comment:
Generally dont want to actually use UnstructuredContent() when passing to one of these methods. If it creates a new Object map, it doesn't actually set it on the Unstructure.Object. Just use toObj.Object. If you're worried about it not being set, you have to do `toObj.Object = toObj.UnstructuredContent()` beforehand

Reply:
Done